src/controllers/OTPcontroller.ts

```diff
@@ -0,0 +1,89 @@
+import { Request, Response } from 'express';
+import { config } from 'dotenv';
+import jwt from 'jsonwebtoken';
+import { generateOtpToken } from '../middleware/otpauthMiddleware';
+import { User } from '../database/models/user';
+import { sendResponse } from '../utils/response';
+import { userToken } from '../utils/tokenGenerate';
+import Database from '../database';
+import { signToken } from '../utils/jwtToken';
+
+config();
+
+export interface DataInfo {
+  body: {
+    id: string;
+    email: string;
+    otp: string;
+  };
+}
+
+export interface UserData {
+  dataValues: {
+    roleId: string;
+    firstName: string;
+    lastName: string;
+    email: string;
+    id: string;
+  };
+}
+
+const sendOtp = async (req: Request, res: Response, isEmail: string) => {
+  const user = (await User.findOne({
+    where: {
+      email: isEmail,
+    },
+  })) as unknown as UserData;
+
+  const { id, email } = user.dataValues;
+
+  const token = await generateOtpToken(id, email);
+
+  return sendResponse<string>(
+    res,
+    201,
+    token,
+    'Verify OTP sent to your email to continue'
+  );
+};
+
+const sellerOtp = async (req: Request, res: Response) => {
+  const info = req.user as DataInfo;
+
+  const tokenSeller = jwt.sign(
+    { id: info.body.id },
+    process.env.JWT_SECRET as string
+  );
+
```

Comment:
wrong implementation of assigning token we have it already

Reply:
This is the token you get after you verify otp and  function for generating otp is not flexible because it only accept 2 arguments but here i have 3 arguments and i don't want to change the codes of others.